pkg/controllers/cloud/cloudprovideraccount_controller.go

```diff
@@ -93,6 +114,18 @@ func (r *CloudProviderAccountReconciler) processDelete(namespacedName *types.Nam
 	if err != nil {
 		return err
 	}
+	err = cloudInterface.DeleteInventoryPoller(namespacedName)
+	if err != nil {
+		return err
+	}
+
+	r.removeAccountPoller(namespacedName)
```

Comment:
Shouldn't you delete account poller(there by closing the goroutine to get cachedVPCs), before calling the DeleteInventortPoller?

Reply:
I was going back and forth on the order, thought again about it. Now I am thinking about deleting account poller first. Reason being, account poller reads from service configs and service configs(including account configs) are cleared when we invoke DeleteInventoryPoller.
However, reversing the order shouldn't affect either, if service configs don't exist account poller won't read anything. Eventually account poller also will get deleted, hence I don't see an issue.